components/eventing-controller/pkg/handlers/nats_test.go

```diff
@@ -4,26 +4,81 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"log"
 	"testing"
 	"time"
 
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
-
 	"github.com/avast/retry-go/v3"
-	cev2event "github.com/cloudevents/sdk-go/v2/event"
+	cenats "github.com/cloudevents/sdk-go/protocol/nats/v2"
+	ce "github.com/cloudevents/sdk-go/v2"
+	cebinding "github.com/cloudevents/sdk-go/v2/binding"
+	ceevent "github.com/cloudevents/sdk-go/v2/event"
 	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
-	"github.com/nats-io/nats.go"
-	. "github.com/onsi/gomega"
-
 	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
 	"github.com/kyma-project/kyma/components/eventing-controller/logger"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/applicationtest"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/fake"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
 	eventingtesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	"github.com/nats-io/nats.go"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	Structured SendEncoding = 0
+	Binary     SendEncoding = 1
 )
 
+type SendEncoding uint8
+
+// SendCloudEventToNATS sends a CloudEvent to a NATS client using a defined encoding method.
+func SendCloudEventToNATS(natsClient *Nats, cloudEvent *ceevent.Event, encoding SendEncoding) error {
+	// validate the cloudEvent
+	if err := cloudEvent.Validate(); err != nil {
```

Comment:
the `ceClient.Send()` from line 63 also validates the cloudEvent, so I guess you can omit this check.